src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
-def measurement_wfm_length(request):
-    MeasWfmLength = collections.namedtuple('MeasurementWaveformLength', ['passed_in', 'expected'])
-    return MeasWfmLength(passed_in=request.param[0], expected=request.param[1])
-
-
-def test_fetch_array_measurement(multi_instrument_session, measurement_wfm_length):
-    test_voltage = 1.0
-    test_record_length = 1000
-    test_num_channels = 2
-    test_meas_wfm_length = measurement_wfm_length.passed_in
-    test_array_meas_function = niscope.ArrayMeasurement.ARRAY_GAIN
-    test_starting_record_number = 2
-    test_num_records_to_acquire = 5
-    test_num_records_to_fetch = test_num_records_to_acquire - test_starting_record_number
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records_to_acquire, True)
-
-    with multi_instrument_session.initiate():
-        waveforms = multi_instrument_session.channels[test_channels].fetch_array_measurement(
-            array_meas_function=test_array_meas_function,
-            meas_wfm_size=test_meas_wfm_length,
-            relative_to=niscope.FetchRelativeTo.PRETRIGGER,
-            offset=5,
-            record_number=test_starting_record_number,
-            num_records=test_num_records_to_fetch,
-            meas_num_samples=2000,
-            timeout=hightime.timedelta(seconds=4))
-
-    assert len(waveforms) == test_num_channels * test_num_records_to_fetch
-    expected_channels = test_channels.split(',') * test_num_records_to_fetch
-    expected_records = [2, 2, 3, 3, 4, 4]
-    for i in range(len(waveforms)):
-        assert len(waveforms[i].samples) == measurement_wfm_length.expected
-        assert waveforms[i].channel == expected_channels[i]
-        assert waveforms[i].record == expected_records[i]
-
-
-def test_fetch_array_measurement_defaults(multi_instrument_session):
-    test_voltage = 1.0
-    test_record_length = 1000
-    test_num_channels = 2
-    test_num_records = 3
-    test_array_meas_function = niscope.ArrayMeasurement.ARRAY_GAIN
-
-    multi_instrument_session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
-    multi_instrument_session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
-
-    with multi_instrument_session.initiate():
-        waveforms = multi_instrument_session.channels[test_channels].fetch_array_measurement(
-            array_meas_function=test_array_meas_function)
-
-    assert len(waveforms) == test_num_channels * test_num_records
-    for i in range(len(waveforms)):
-        assert len(waveforms[i].samples) == test_record_length
-
-
-@pytest.fixture(params=[numpy.int8, numpy.int16, numpy.int32, numpy.float64])
-def fetch_waveform_type(request):
-    return request.param
-
-
-def test_fetch_into(multi_instrument_session, fetch_waveform_type):
```

Comment:
`fetch_into` isn't supported on grpc-device, so this test was moved to just be in the `TestLibrary` class below.

Reply:
It would be nice to add a comment above the test.
"# not supported by grpc due to numpy usage"